store/pogreb/pogreb.go

```diff
@@ -94,101 +95,50 @@ func (s *pStorage) RemoveProvider(providerID peer.ID) error {
 	}
 	iter := s.store.Items()
 
-	valsToDel := map[string]struct{}{}
-
-	s.mdLock.Lock()
-	defer s.mdLock.Unlock()
+	s.valLock.Lock()
+	defer s.valLock.Unlock()
 
 	for {
-		key, val, err := iter.Next()
+		key, valueData, err := iter.Next()
 		if err != nil {
 			if err == pogreb.ErrIterationDone {
 				break
 			}
 			return err
 		}
 
-		if !bytes.HasPrefix(key, indexKeyPrefix) {
-			// Key does not have index prefix, so is not an index key.
+		if !bytes.HasPrefix(key, valueKeyPrefix) {
+			// Key does not have value prefix, so is not an value key.
 			continue
 		}
 
-		values, err := indexer.UnmarshalValues(val)
-		if err != nil {
-			return err
-		}
-
-		if len(values) == 0 {
-			if err = s.store.Delete(key); err != nil {
-				return err
-			}
-			continue
-		}
-
-		// Separate values into those to remove and those to keep.
-		var vdel, vkeep []indexer.Value
-		for i := range values {
-			if values[i].ProviderID == providerID {
-				vdel = append(vdel, values[i])
-			} else {
-				vkeep = append(vkeep, values[i])
-			}
-		}
-
-		if len(vdel) == 0 {
-			continue
-		}
-
-		// If there are no values to keep, then remove the index.  Otherwise,
-		// update the index to have the remaining values.
-		if len(vkeep) == 0 {
-			if err = s.store.Delete(key); err != nil {
-				return err
-			}
-		} else {
-			// store the list of value keys for the multihash
-			b, err := indexer.MarshalValues(vkeep)
+		if valueData != nil {
+			value, err := indexer.UnmarshalValue(valueData)
 			if err != nil {
 				return err
 			}
-			err = s.store.Put(key, b)
-			if err != nil {
-				return err
-			}
-		}
 
-		for i := range vdel {
-			valsToDel[string(vdel[i].ContextID)] = struct{}{}
+			if value.ProviderID != providerID {
```

Comment:
keep a metric of how many of these we're seeing?

Reply:
I added a metric to track the number of provider deletes.  If that is not what you are looking for, LMK.